src/block-management/ethrpc-transport-adapter.js

```diff
@@ -0,0 +1,41 @@
+"use strict";
+
+var nextToken = 1;
+var subscriptionMapping = {};
+
+module.exports = function (ethrpc) {
+  return {
+    getBlockByNumber: ethrpc.getBlockByNumber.bind(ethrpc),
+    getBlockByHash: ethrpc.getBlockByHash.bind(ethrpc),
+    subscribeToReconnects: ethrpc.internalState.transporter.addReconnectListener.bind(ethrpc.internalState.transporter),
+    unsubscribeFromReconnects: ethrpc.internalState.transporter.removeReconnectListener.bind(ethrpc.internalState.transporter),
+    subscribeToNewHeads: function (onNewHead, onSubscriptionError) {
+      var token = (nextToken++).toString();
+      subscriptionMapping[token] = null;
+      try {
```

Comment:
The try/catch is defensive against external (not part of this class) code throwing for some reason.  However, you are right that it is inconsistent with the rest of this project (which isn't written defensively) and even the rest of the code in this PR and the previous, so I have removed it.